### PR TITLE
fix loss normalization with entropy mask

### PR DESCRIPTION
**Pull request by hjh0119**
# What does this PR do?

Fix a bug where, when using entropy_mask, tokens that were filtered out would still be included in loss normalization even though their loss was set to zero.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/trl/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [ ] Was this discussed/approved via a GitHub issue? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Comment by kashif** [comment]
@hjh0119 if we only mask the `completion_mask` then I believe the KL will only be computed on the high entropy tokens and the clip ratio as well... and other issues... i think it's best to do:

```python
        if entropy_mask is not None:
            per_token_loss = per_token_loss & entropy_mask
            # Combine entropy_mask with completion_mask for proper normalization
            effective_mask = completion_mask & entropy_mask
        else:
            effective_mask = completion_mask
        if self.beta != 0.0:
            per_token_loss = per_token_loss + self.beta * per_token_kl

        if self.loss_type == "grpo":
            loss = ((per_token_loss * effective_mask).sum(-1) / effective_mask.sum(-1).clamp(min=1.0)).mean()
        elif self.loss_type == "bnpo":
            loss = (per_token_loss * effective_mask).sum() / effective_mask.sum().clamp(min=1.0)
        elif self.loss_type == "dr_grpo":
            loss = (per_token_loss * completion_mask).sum() / (per_token_loss.size(0) * self.max_completion_length)
        else:
            raise ValueError(f"Unknown loss type: {self.loss_type}")
```

as this isolates entropy masking to only the loss computation while preserving all other functionality

can you try this out?

**Comment by qgallouedec** [comment]
I don't think it's an issue actually. We have discussed this question here: https://github.com/huggingface/trl/issues/3709#issuecomment-3064518856

**Comment by hjh0119** [comment]
That makes sense, I’ll close this PR.
